### PR TITLE
Improve formatting

### PR DESCRIPTION
**Pull request by domhof**
# Purpose

Describe the purpose of the pull request.
(Long descriptions should be in documentation, not here)

# Checklist

Please only check relevant points.
For docu fixes, spell checking and similar nothing
needs to be checked.

- [ ] commit messages are fine (with references to issues)
- [ ] I ran all tests and everything went fine
- [ ] I added unit tests
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions
- [ ] meta data is updated (e.g. README.md of plugins)

@markus2330 please review my pull request

### CODE REVIEW

**Comment by markus2330** [comment]
Thanks for creating the PR!

It is a good idea to add fences, but why not saying which language is used? This improves syntax highlighting.

And in `doc/CODING.md` (Markdown Guidelines) we decided to use the `#` of titles on the left and right side, please revert these changes.

**Comment by domhof** [comment]
Thanks for the feedback! I've added the language specifiers. What is the benefit of having the `#` on both sides of the title?

**Comment by markus2330** [comment]
> What is the benefit of having the # on both sides of the title?

It is only a matter of taste, I also had it mostly done as you propose. But we decided to do it that way as documented and reformatted everything consistently thereafter. So the benefit now is consistency.

**Comment by domhof** [comment]
Okay, I've added the '#' as well. However, since markdown neither enforces it nor does it check if it is the same number of '#' on both sides, I would argue that duplicating the '#' on the right side is only a potential source of error in terms of consistency. So for this reason I would be happy if we could rethink this decision in order to keep things simple and tidy (after all, it will have an impact also on future documents for a long time to come).

File: doc/decisions/high_level_api.md [hunk 1]
```diff
@@ -102,11 +134,13 @@ What is not so nice:
 ## Argument ##
 
 1. Very easy to get started with, to get a key needs 3 lines of codes:
-   ```c
-   KDBHL *handle = kdbhlOpen ("/sw/elektra/kdb/#0/current");
-   printf ("number /mykey is %d\n", kdbhlGetInt (handle, "/mykey"));
-   kdbhlClose (handle);
-   ```
+   
+```c
```
Comment (markus2330) [inline doc/decisions/high_level_api.md]: I am not sure if every markdown renderer will handle this as expected. Some might start counting again from `1.` below? Not really important, so I will merge anyway.

**Comment by markus2330** [comment]
Thank you! For future commits, please use `module: ...` syntax in the commit messages, where module in this case would be `doc:`.

> However, since markdown neither enforces it nor does it check if it is the same number of '#' on both sides, I would argue that duplicating the '#' on the right side is only a potential source of error in terms of consistency. So for this reason I would be happy if we could rethink this decision in order to keep things simple and tidy (after all, it will have an impact also on future documents for a long time to come).

When we had the requirement the other way round, it was inconsistent, too (much more than it is now). I never saw the case that the number of `#` on the left and right side were differently. So I would argue it really does not matter at all.

But if it is important to you, please create a proposal for a "better" markdown style. But please be aware that reformatting the current base is a considerable effort. Hopefully we will find a style-formatter for Markdown soon, that would be the easiest solution. (We tried some, see #1034, but they were not configurable and we did not find time to investigate them further.)

**Pull request merged**